src/economy/portdock.cc

```diff
@@ -162,10 +159,13 @@ bool PortDock::init(EditorGameBase& egbase) {
  * that we merge with a larger fleet when possible.
  */
 void PortDock::init_fleet(EditorGameBase& egbase) {
+	assert(!fleet_);
 	ShipFleet* fleet = new ShipFleet(get_owner());
 	fleet->add_port(egbase, this);
 	fleet->init(egbase);
 	// Note: the Fleet calls our set_fleet automatically
+	assert(fleet_);
+	fleet_->update(egbase);
 }
 
 void PortDock::cleanup(EditorGameBase& egbase) {
```

Comment:
Please add acomment: what is cleaned up? ist this a kind of seconadry DTor (Destructor)?

Reply:
I did not touch that method at all…

MapObject::cleanup() (which is overridden by every MapObject subclass) is called directly before destruction so that DTor-actions that require an EditorGameBase can be performed.

Reply:
OK, thats the comment I had expected, please add it.